packages/expo-modules-autolinking/src/autolinking/__tests__/findModules-test.ts

```diff
@@ -430,7 +424,69 @@ describe(findModulesAsync, () => {
     });
 
     expect(result).toEqual({
-      '@expo/expo-test': expect.objectContaining({}),
+      '@expo/expo-test': expectAnyModule(),
+    });
+  });
+
+  /**
+   * TODO(@kitten): Documentation implies this should work
+   * /app
+   *   ├── /app/local-expo-module
+   */
+  it.skip('should link local non-node module (FAILURE)', async () => {
+    vol.fromNestedJSON(
+      {
+        ...mockedRoot(),
+        'local-expo-module': mockedModule('react-native-third-party'),
+      },
+      projectRoot
+    );
+
+    const result = await findModulesAsync({
+      searchPaths: [path.join(projectRoot, 'node_modules')],
+      platform: 'ios',
+      projectRoot,
+    });
+
+    expect(result).toEqual({
+      'local-expo-module': expectAnyModule(),
+    });
+  });
+
+  /**
+   * TODO(@kitten): Fix this behaviour
+   * /app
+   *   ╚══ /node_modules/expo-module-1
+   *       └── /app/node_modules/expo-module-1/node_modules/expo-module-2
+   */
+  it('should link transitive non-hoisted module (FAILURE)', async () => {
```

Comment:
This, I expected to pass but doesn't. I know that some code basically resolves by dependencies by looking at `pkg.dependencies` (we'll likely have to extend this for `peerDependencies` since these can auto-install, and deduplicate target paths).

But that code/logic might be fighting the non-structural / non-traversing scanning logic. Something bails early and doesn't let this sub-dependency be discovered.

This is probably legacy behaviour that assumes that hoisting is always active